tests/Updater/PuliStrategyTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the puli/cli package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Cli\tests\Updater;
+
+use PHPUnit_Framework_Assert;
+use PHPUnit_Framework_TestCase;
+use Puli\Cli\PuliApplicationConfig;
+use Puli\Cli\Updater\PuliStrategy;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+class PuliStrategyTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var PuliStrategy
+     */
+    private $strategy;
+
+    public function setUp()
+    {
+        $this->strategy = new PuliStrategy();
+    }
+
+    public function testDownload()
+    {
+        $pharTemporaryFile = __DIR__.'/tmp.phar';
```

Comment:
You should not create temporary files in the same directory. Otherwise they easily get committed to Git by accident.

Instead, create a temporary directory in the system's temp directory. Make sure to remove that directory again in `tearDown()` (not in the test, which might fail).

See `PuliBinTest` for an example.